### ISSUE TITLE
`reportIncompatibleVariableOverride` false positive on overriden descriptor attribute with the same return type as the descriptor's `__get__`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1112]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAECCBcBYAUNN0AmBTAZtA%2BvgOZYAuhAFBFiDgDTQCWAdhKWM8FvNAPYBGAKyzBSDXgHdmWAE49SATwAOWANoDhogLrQAPtAByvadAC8h41gYB6AJTQAtAD4mzUklTovAOl8oUoJAwAEJgciheYDywZnAUtv7IgVDQoQBeFKEytjwR6FGupLEAjNAAxNCyMrwyQA)

```python
class A:
    def __get__(self, instance: object, owner: type[object] | None = None, /) -> int:
        ...

class Bar:
    a: A = A()

class Baz(Bar): 
    a: int = 1 # error
```

```
"a" overrides symbol of same name in class "Bar"
  Variable is mutable so its type is invariant
    Override type "int" is not the same as base type "A" (reportIncompatibleVariableOverride)
```

i cant think of a reason this would be unsafe, so it should be allowed

**Comment by DetachHead** [comment]
possibly related: #1025
